pkg/crc/preflight/preflight_checks_common.go

```diff
@@ -144,3 +148,30 @@ func removeOldLogs() error {
 	}
 	return nil
 }
+
+func pullBundleImage(preset preset.Preset) error {
+	imgHandler := image.OCIImageHandler{
+		ImageName: image.GetImageName(preset),
+		ImageTag:  version.GetCRCVersion(),
+		Progress:  os.Stdout,
+	}
+	destDir, err := ioutil.TempDir(constants.MachineCacheDir, "")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(destDir)
+	imgDigest, err := imgHandler.CopyImage(destDir)
+	if err != nil {
+		return err
+	}
+	bundleLayerPath, err := image.GetBundleImageLayerFromManifest(imgDigest)
+	if err != nil {
+		return err
+	}
+	logging.Info("Extracting the image bundle layer...")
+	_, err = extract.Uncompress(filepath.Join(destDir, bundleLayerPath), constants.MachineCacheDir, true)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
Wondering if this deserves to be a helper in the `image` package as this is quite a few low-level steps that needs to be done, not sure preflight should have that much knowledge about how a container image is stored.